packages/docs/build/bots/onchain-integrations.mdx

```diff
@@ -0,0 +1,374 @@
+---
+title: Onchain Integrations
+---
+
+Towns bots are onchain apps with two blockchain capabilities: direct contract interactions using your bot's wallet, and requesting users to sign transactions or messages.
+
+## Bot Wallet Architecture
+
+Your bot has two addresses:
+- **Gas wallet** (`bot.viem.account`) - Signs and pays for transactions
+- **App address** (`bot.appAddress`) - Treasury wallet (SimpleAccount)
+
+<Note>
+Your gas wallet needs Base ETH for transaction fees. See [Getting Started](/build/bots/getting-started#understanding-your-bots-wallet-architecture) for details.
+</Note>
+
+## Configuration
+
+### Base RPC URL
+
+For reliable blockchain interactions, configure a custom RPC endpoint. The default public RPC has strict rate limits.
+
+```typescript
+const bot = await makeTownsBot(
+  process.env.APP_PRIVATE_DATA!,
+  process.env.JWT_SECRET!,
+  {
+    baseRpcUrl: process.env.BASE_RPC_URL,
+    commands
+  }
+)
+```
+
+Add to `.env`:
+
+```bash
+BASE_RPC_URL=https://base-mainnet.g.alchemy.com/v2/YOUR_API_KEY
+```
+
+Get a free RPC URL from [Alchemy](https://www.alchemy.com/) or similar providers.
+
+## Bot Contract Interactions
+
+### Reading from Contracts
+
+Read contract state without gas costs:
+
+```ts
+import { readContract } from 'viem/actions'
+import simpleAppAbi from '@towns-protocol/bot/simpleAppAbi'
+
+const owner = await readContract(bot.viem, {
+  address: bot.appAddress,
+  abi: simpleAppAbi,
+  functionName: 'moduleOwner',
+  args: []
+})
+```
+
+### Writing to Contracts
+
+Use `execute` from ERC-7821 for any onchain interaction:
+
+```ts
+import { execute } from 'viem/experimental/erc7821'
+import { parseEther } from 'viem'
+import { waitForTransactionReceipt } from 'viem/actions'
+
+const hash = await execute(bot.viem, {
+  address: bot.appAddress,
+  account: bot.viem.account,
+  calls: [{
+    to: tokenAddress,
+    abi: erc20Abi,
+    functionName: 'transfer',
+    args: [recipientAddress, parseEther('1.0')]
+  }]
+})
+
+await waitForTransactionReceipt(bot.viem, { hash })
+```
+
+### Batch Transactions
+
+Execute multiple operations atomically:
+
+```ts
+const hash = await execute(bot.viem, {
+  address: bot.appAddress,
+  account: bot.viem.account,
+  calls: [
+    {
+      to: tokenAddress,
+      abi: erc20Abi,
+      functionName: 'approve',
+      args: [dexAddress, amount]
+    },
+    {
+      to: dexAddress,
+      abi: dexAbi,
+      functionName: 'swap',
+      args: [tokenIn, tokenOut, amount]
+    }
+  ]
+})
+```
+
+All operations succeed or all fail.
+
+## User Interactions
+
+Request users to sign transactions or messages from their wallets.
+
```

Comment:
We should explain what user interaction are.

Also mention that it allow the user to send custom input/buttons.
IMO it should be in a dedicated page. That's why we choose to rename onchain/external interactions to integrations, so we have the interaction name free to have a new page.